src/top/utop.el

```diff
@@ -595,21 +615,43 @@ it is started."
                              (cadr (split-string prefix "\\."))
                            prefix)))
              (when (string-prefix-p prefix argument)
-              (push argument utop-completion)
-              (throw 'done t))))))
+               (push argument utop-completion)
+               (throw 'done t))))))
       ;; End of completion
       ("completion-stop"
        (utop-set-state 'edit)
-       (if (utop--supports-company)
-           (funcall utop--complete-k (nreverse utop-completion))
+       (cond
+        ((utop--supports-company)
+         (funcall utop--complete-k (nreverse utop-completion)))
+        (utop-capf-enable
+         (setq utop-capf-completion (nreverse utop-completion)))
+        (t
          (progn
            (if (> (length utop-completion) 1)
                (with-current-buffer utop-complete-buffer
                  (with-output-to-temp-buffer "*Completions*"
                    (display-completion-list (nreverse utop-completion))))
-             (minibuffer-hide-completions))))
+             (minibuffer-hide-completions)))))
        (setq utop-completion nil)))))
 
+(defun utop-completion-at-point ()
+  "Complete thing at point."
+  (when utop-capf-enable
+    (setq utop-capf-completion nil)
```

Comment:
Seems this name is misleading as that's actually the list of completion candidates, right?